src/zmalloc.c

```diff
@@ -103,17 +103,25 @@ static void (*zmalloc_oom_handler)(size_t) = zmalloc_default_oom;
 
 /* Try allocating memory, and return NULL if failed.
  * '*usable' is set to the usable size if non NULL. */
+/*
+ * 尝试分配空间, 如果分配失败返回 NULL .
+ * 如果分配成功, 还会把 *usable 设置为可用空间的大小.
+ *
+ * ( 建议先看 zmalloc 函数的注释 )
+ * 首先分配一个 size + sizeof(size_t) 大小的空间
+ * 然后判断宏 HAVE_MALLOC_SIZE 是否存在, 也就是是否存在 zmalloc_size 函数, 然后进行具体逻辑.
+ */
 void *ztrymalloc_usable(size_t size, size_t *usable) {
     ASSERT_NO_SIZE_OVERFLOW(size);
     void *ptr = malloc(MALLOC_MIN_SIZE(size)+PREFIX_SIZE);
 
     if (!ptr) return NULL;
-#ifdef HAVE_MALLOC_SIZE
+#ifdef HAVE_MALLOC_SIZE  /* 存在 zmalloc_size 函数, 可以直接调函数获取实际 malloc 分配的空间 */
```

Comment:
我们换个新行来加注释把，就尽量不改动它原本的代码行，减少 diff 啥的，以后追踪代码变更，或者合并新 redis 代码也能减少冲突，后面的也都类似
```suggestion
#ifdef HAVE_MALLOC_SIZE
/* 存在 zmalloc_size 函数, 可以直接调函数获取实际 malloc 分配的空间 */
```